documentation/assemblies/assembly-sizing-guide.adoc

```diff
@@ -0,0 +1,309 @@
+[id='sizing-guide-{context}']
+= Sizing guide
+
+The sizing guide provides guidelines on how to size {ProductName} installations and what tradeoffs you can expect when adjusting the configuration. Sizing {ProductName} involves configuration of:
+
+* Brokers 
+* Routers (standard address space only)
+* Operator(s)
+* Plans
+
+Each address space type have some distinct features that should be taken into account when creating the address plans. For more information about address space types and their semantics, see link:{BookUrlBase}{BaseProductVersion}{BookNameUrl}#con-address-space-messaging[address spaces].
+
+NOTE: Sizing {ProductName} components also requires that your {KubePlatform} cluster has sufficient capacity to handle the requested resources. If the {KubePlatform} nodes are configured with 4GB of memory, you cannot configure brokers and routers with memory sizes larger than. Also, as each address space creates a dedicated piece of infrastructure, you need to ensure that cluster capacity can meet demand as the number of address spaces increases. Finally, the use of affinity and tolerations may also restrict the nodes available for the messaging infrastructure to use.
+
+== Sizing brokers
+
+Brokers are configured in the `BrokeredInfraConfig` and `StandardInfraConfig` resources. When sizing a broker, the following must be taken into consideration:
+
+* The average message size
+* The number of messages stored
+* The number of queues and/or topics
+* The address full policy
+
+NOTE: In {ProductName} it is only possible to restrict the total number of memory allocated for a broker, and not restrict memory used by individual addresses.
+
+Enabling address full policy `PAGE` will allow you to store more messages than can fit in memory, at the expense of a potential performance hit from reading data from disk. Therefore, paging is mainly of interest when you have large messages or expect a long backlog of messages in your system.
+
+=== Example
+
+Given 10 queues with a maximum of 1000 messages stored per queue, an average message size of 128kB, the amount of storage space required to store messages is:
+
+```
+10 * 1000 * (128 * 1024) = 1.25 GB
+```
+
+In addition, the broker has a fixed storage footprint of about 50MB.
+
+The amount of memory required for the broker depends on which address full policy is being used. If the PAGE policy is used, the memory requirements can be reduced by having messages stored on disk rather than in the journal (which always needs to fit in memory). If the FAIL, BLOCK or DROP policies are being used, all messages must be held in memory.
```

Comment:
I know the point you are making, but I think a reader may worry that their persistent messages aren't being written to non-volatile storage.

Reply:
@k-wall Agreed. I've rephrased this and a similar comment above.